### DISCUSSION TITLE
Quick_11_iso_esd_wim_TPM_toggle on WIM custom

### DISCUSSION THREAD

**Discussion by S7repo** [AveYo/MediaCreationTool.bat#88] [General]
Hello,

I thank @AveYo for its work. However, I have a problem.

Indeed, I created a custom Windows 11 ISO with Sysprep which works perfectly via bootable USB key and I now want to make it 'Upgrade' compatible without key.

So I ran the Quick_11_iso_esd_wim_TPM_toggle.bat script which finished without error but the upgrade mode does not work in direct execution.

I have the following error:

Windows Server installer
The installer failed to validate the product key
![image](https://user-images.githubusercontent.com/33670742/149743157-8fff0a79-3710-4a44-bc7a-86386c0e1d5a.png)

Can you help me?

**Comment by AveYo** [comment]
What kind of media you've made (what editions)?
Did you add a `PID.txt` to the \sources?

Did you add a generic `EI.cfg` to `\sources` with content:
`[Channel]`
`_Default`

or if that one does not work:
`[Channel]`
`_Default`
`[VL]`
`0`

**Comment by S7repo** [comment]
Hi AveYo,

Thank you for your quick answer.
I only did this:

1- Extracted Windows 11 Pro from the official Microsoft ISO with the command
`Dism /Get-WimInfo /WimFile:"c:\sources\install.esd"`
and
`Dism /export-image /SourceImageFile:c:\sources\install.wim /SourceIndex:6 /DestinationImageFile:c:\iso\install.wim /Compress:max /CheckIntegrity`

2- Importing install.wim(Windows 11 Pro) into the ISO with PowerISO

3- Mount a virtual machine with VMWARE and install the software, settings....

4- Compile with SYSPREP and create install.wim with the following command: 
`dism /Capture-Image /ImageFile:E:\install.wim /CaptureDir:C:\ /Name:"blabla"`

**Comment by AveYo** [comment]
Add the generic `EI.cfg` to \sources folder and try again. 
It definitely works with untouched iso, so if it still does not work, there must be some step you've overlooked with your sysprep

**Comment by S7repo** [comment]
I just took the official ISO and right click "send to" and "Quick_11_iso_esd_wim_TPM_toggle.bat"

TPM script OK but setup.exe doesn't run anymore

**Reply by AveYo** [reply]
Works fine here, must be something on your PC.

How exactly does not run anymore?
Can you make a capture?
Can you provide your `C:\$WINDOWS.~BT\Sources\Panther` log folder?

**Comment by S7repo** [comment]
Ok. here is a video is the log file
[(https://drive.infomaniak.com/app/share/240972/68ce1ddf-8f04-428a-b21e-63c66251942c)](url)

**Comment by S7repo** [comment]
Yes; Logs + video
[For AveYo.zip](https://github.com/AveYo/MediaCreationTool.bat/files/7884434/For.AveYo.zip)

**Comment by AveYo** [comment]
InitializePaths(3410): Result = 0x8007000D
Something is blocking setup from writing the temporary files.
Reboot.
Delete `C:\$WINDOWS.~BT` and `C:\$Windows.~WS` (hidden by default).
Try again.

**Comment by S7repo** [comment]
No it doesn't work after reboot and deleting the directories `C:\$WINDOWS.~BT` and `C:\$Windows.~WS`
I also tested on another PC installed in official Windows 10 Pro and it's the same.

**Comment by S7repo** [comment]
[Panther.zip](https://github.com/AveYo/MediaCreationTool.bat/files/7887543/Panther.zip)

**Comment by AveYo** [comment]
I and others also tested on multiple PCs and it works as advertised.
Even if I knew it's a waste of time, I have downloaded the `Win11_French_x64v1.iso` and followed your steps anyway. 
Script worked fine, bypass worked fine, setup worked fine.
It's a problem on your end. 
Maybe the shitty PowerISO with it's shell extension & iso mount hijack or whatever else you have there is doing something bad. 
Or maybe you made a mistake and did not even add the install.wim in the sources folder (should have received a replace dialog).

Simplest thing is to extract everything from the official iso to the hdd and run setup from there. 
It should launch and it should complain about tpm if you're doing it on an unsupported PC.
Then can browse to the sources folder and right-click install.wim - Send to - Quick..
Now when running setup it should say Windows Server instead of 11 - that's how you know the bypass is active.
Can add a PID.txt with the key or the generic ei.cfg I've mentioned in the sources folder to bypass prompt for key.
One last thing, you can toggle the bypass at any step, either on the install.wim / install.esd if the files are extracted on the hard disk, or on the iso (before / after you do your authoring / poweriso thing - it does not matter)

**Comment by S7repo** [comment]
Ok thanks I'll try on my side.
Thanks for your help. It's nice

**Comment by S7repo** [comment]
Hi AveYo,
Here I have reproduced the test on a Windows 10 Pro VM.
I have indeed a problem with my PC and I will see later.

On the VM, I tried the official ISO and patch with Quick_11_iso_esd_wim_TPM_toggle.bat
It works.

Now I did this test with my ISO edited with Sysprep. Patch `Quick_11_iso_esd_wim_TPM_toggle.bat` OK but run with error
**Windows Server installer
The installer failed to validate the product key**

In short, as soon as install.wim is edited with Sysprep, I get a "Windows Server installer
The installer failed to validate the product key".

I wonder how to fix this. Can you tell me what the script Quick_11_iso_esd_wim_TPM_toggle.bat modifies in install.wim?
With your answer, maybe I can find a solution to my problem

Thank you very much for your help

**Comment by AveYo** [comment]
It just modifies one parameter on the image description, switching it from CLIENT to SERVER.
Setup reads that and acts accordingly. 
This does make a difference, in that you cannot skip the key with "enter later" like on client, instead setup asks for a key before install, and you must either add a generic ei.cfg, set the key via PID.txt or set the key via unnatend.xml / autounnatend.xml to hide the prompt

your issue must be related to the product key, 
either wrong one inserted or a lack of one in the sysprep-ed registry
I have no idea how you do your sysprep and what could go wrong

have you tried adding the generic ei.cfg to sources folder?
how about a PID.txt with the relevant key to sources folder?
setup tries to find a suitable product key in those files, else it searches in the registry

I assume there is a unnatend.xml / autounnatend.xml file used on the media?
that also would require a key entry to be inserted in it
I think you need to read more about `"windows sysprep product key"` - [for example this](https://social.technet.microsoft.com/Forums/windows/en-US/4104fa3f-9c36-4d45-aa36-677602894768/sysprep-maintain-activation-and-product-key)

**Comment by S7repo** [comment]
Here is the exact command for sysprep that I use.
`sysprep.exe /oobe /generalize /unattend:unattend.xml /shutdown`

I tried with the generic `ei.cfg` and `PID.txt` whose content is as follows:

```
[PID]
Value=VK7JG-NPHTM-C97JM-9MPGT-3V66T
```

Do you think this generic key is valid?

I also have this product key in my `unattend.xml` file here (txt extension for Github)
[unattend-clients.txt](https://github.com/AveYo/MediaCreationTool.bat/files/7913147/unattend-clients.txt)

**Comment by AveYo** [comment]
The xml key seems fine, but maybe something else intervenes, 
you should at least read the topic I linked, or find more similar issues  with the keywords.
I only skimmed through it, there were some quirks like needing to pre-activate when doing sysprep, or else do not add a key and let the user input it on the next real boot.
 
I would try a dummy key entry first:
```xml
            <UserData>
                <ProductKey>
                    <Key>AAAAA-VVVVV-EEEEE-YYYYY-OOOOO</Key>
                    <WillShowUI>OnError</WillShowUI>
                </ProductKey>
            </UserData>

```
then the KMS generic one: W269N-WFGWX-YVC9B-4J6C9-T83GX
then remove the whole UserData section in your unnatend.xml.
Notice I've used the `WillShowUI` entry.

Sorry, I do not have the time to dive more into it atm.

**Comment by S7repo** [comment]
Ok thanks for the feedback.
So I did this in the unattend. Is this good for you before I start testing?
![image](https://user-images.githubusercontent.com/33670742/150700007-5d7acb04-bacf-4d45-82cd-b7e8c42039fd.png)

**Comment by S7repo** [comment]
Hello,

For the test, I took the clean ISO from the microsoft website and I installed a new VM.

I only installed notepad++.

Then I restarted my VM on the installation mode and entered the command:
`dism /Capture-Image /ImageFile:E:\install.wim /CaptureDir:C:\ /Name: "toto"`
then 
patch in the script `Quick_11_iso_esd_wim_TPM_toggle.bat`
The problem is still present. So we can delete the sysprep theory

**Reply by AveYo** [reply]
I still don't understand what are you trying to achieve.
Capture-image is meant to make an image of an **installed** OS, to be used later via apply-image, **bypassing install setup** altogether.
Not for the setup image itself, not to run install setup later.

What you're trying seems to sysprep setup image itself, then run setup again?!
What you should do is install normally, then sysprep the resulting installation (C:), capture-image & etc. and finally apply-image via a boot.wim unnatend.xml / script. 
You can't just run setup with that sysprep'ed image.

**Comment by S7repo** [comment]
Yes
1- I install the system
2- I add the software
3- I recreate the wim

The goal is to create an installation of Windows 11 with my pre-installed software

Unless you have a better solution, I'm interested

**Reply by AveYo** [reply]
Again, you can't just run setup with that sysprep'ed image. 
[You're meant to apply it via a script from within WinPE](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/capture-and-apply-windows-using-a-single-wim
)
That could be automated via boot.wim `winpeshl.ini`

To use the normal installation setup you need a different approach entirely (make use of $OEM$ folders to drop the installed programs, import required registry entries and so forth).

**Comment by shikulja** [comment]
@AveYo 
One more problem, using more recent uupdump builds including updates. The Quick_11_iso_esd_wim_TPM_toggle script works well, but I would like to have a more complete version, with support for AutoUnattend.xml \ EI.cfg, etc.

**Reply by AveYo** [reply]
I did plan to share a script taking all the goodies from MediaCreationTool.bat and apply it to a non-mct iso, guess that's what you're after.

The issue is with mounting & extracting the iso (since I refuse to bundle external binaries like 7z).
Powershell can do it, but it's slow (and not on windows 7 by default).

Much simpler if input is an already extracted folder.
That's doable, and I might do it if you're interested.

**Reply by shikulja** [reply]
This option is suitable, rewriting the image manually will not be too difficult.
